src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectCodeModel.cs

```diff
@@ -49,29 +42,17 @@ private EnvDTE.ProjectItem GetProjectItem(string filePath)
 
         public EnvDTE.CodeModel GetCodeModel(EnvDTE.Project parent)
         {
-            var codeModelCache = GetCodeModelCache();
-            if (codeModelCache == null)
-            {
-                return null;
```

Comment:
this makes it look like GteCodeModelCache can return null.  but it looks like other code isn't resilient to that.

Reply:
See other comment: it could happen, but only if everything is already broken. And since half of the code didn't handle it, it would still probably explode. I suspect this was bad refactorings over time resulted in broken code, and then new code (like this) just added paranoia because well, anything could be null! It's C#!